### ISSUE TITLE
[Storage] Getting error "Could not insert data: invalid character '<' looking for beginning of value" while using batch operations on tables and sasToken client.

### ISSUE THREAD

**Issue by rafalrygas** [Azure/azure-sdk-for-go#14369]
### Bug Report

Paths:
` github.com/Azure/azure-sdk-for-go/storage`

Version: v51.3.0

**What happened?**

After creating storage client using `NewAccountSASClientFromEndpointToken` everything works correctly. 
When trying to create storage client using `NewAccountSASClientFromEndpointToken` there is always an error while using batch operations on tables (`Could not insert data: invalid character '<' looking for beginning of value`). Standard operations work correctly.

**What did you expect or want to happen?**

Working with client created by `NewAccountSASClientFromEndpointToken` method.

**How can we reproduce it?**

1. Create storage account
2. Generate sasToken
3. Create table
4. Run below code

```
import (
	azstorage "github.com/Azure/azure-sdk-for-go/storage"
)

//...

client, _ := azstorage.NewAccountSASClientFromEndpointToken(host, sasToken)

tableService := client.GetTableService()

table := tableService.GetTableReference(tableName)

tableBatch := table.NewBatch()

entity := table.GetEntityReference("partkey1", "rowkey1")

tableBatch.InsertOrReplaceEntityByForce(entity)

err := tableBatch.ExecuteBatch()
println(err.Error())
```

Returns `Could not insert data: invalid character '<' looking for beginning of value`

**Comment by ArcturusZhang** [comment]
Hi @rafalrygas thanks for this issue!

@jhendrixMSFT could you please take a look at this?

**Comment by jhendrixMSFT** [comment]
This is due to attempting to unmarshall an XML response as JSON.  It's been fixed in `v52.3.1`
